scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+        exit 114 # EALREADY
+    fi
+    rm -rf "$CLONE_PID_FILE"
+
+    # If port was not set up explicitly in wsrep_sst_address,
+    # then we need to set the port to a default that is different to the default port existing in the running instance.
+    # This because clone port <> mysql port, due to the need to spin a second MySQL instance able to accomodate the clone action
+    wsrep_log_debug "-> WSREP_SST_OPT_PORT $WSREP_SST_OPT_PORT"
+    wsrep_log_debug "-> WSREP_SST_OPT_ADDR_LOCAL $WSREP_SST_OPT_ADDR_LOCAL"
+
+    IFS=':' read -ra my_array <<< "$WSREP_SST_OPT_ADDR_LOCAL"
+
+    lenAr=${#my_array[@]}
+    wsrep_log_debug "-> lenarray $lenAr"
+
+    if [ $lenAr -gt 1 ]; then
+        JOINER_CLONE_HOST=$(echo $WSREP_SST_OPT_ADDR_LOCAL | cut -d ':' -f 1)
+        JOINER_CLONE_PORT=$(echo $WSREP_SST_OPT_ADDR_LOCAL | cut -d ':' -f 2)
+     else
+        JOINER_CLONE_HOST="$WSREP_SST_OPT_ADDR_LOCAL"
+        JOINER_CLONE_PORT=$CLONE_INSTANCE_PORT
+    fi
+
+
+    wsrep_log_debug "-> JOINER_CLONE_HOST $JOINER_CLONE_HOST"
+    wsrep_log_debug "-> JOINER_CLONE_PORT $JOINER_CLONE_PORT"
+
+
+    # Define the tmp directory
+    tmp_datadir=$(wsrep_mktemp_in_dir "$WSREP_SST_OPT_DATA" -d)
+
+
+    CLONE_ERR="$WSREP_SST_OPT_DATA/$MODULE.err"
+    CLONE_SQL="$WSREP_SST_OPT_DATA/$MODULE.sql"
+    IST_FILE="$WSREP_SST_OPT_DATA/IST_FILE.txt"
+    GRASTATE_FILE="$WSREP_SST_OPT_DATA/grastate.dat"
+
+    CLONE_SOCK="$WSREP_SST_OPT_DATA/recover_clone.sock"
+    CLONE_X_SOCK="$WSREP_SST_OPT_DATA/cmysqlx.sock"
+
+    DONOR_IP=""
+
+
+    if [ ${#CLONE_X_SOCK} -ge 104 ]
+    then
+        wsrep_log_error "Unix socket path name length for CLONE SST receiver"\
+                        "'$CLONE_X_SOCK' is greater than commonly acceptable"\
+                        "limit of 104 bytes."
+                        # Linux: 108, FreeBSD: 104
+        exit $EINVAL
+    fi
+
+    [ -z "$WSREP_SST_OPT_CONF" ] \
+    && DEFAULTS_FILE_OPTION="" \
+    || DEFAULTS_FILE_OPTION="--defaults-file='$WSREP_SST_OPT_CONF'"
+
+    [ -z "$WSREP_SST_OPT_CONF_SUFFIX" ] \
+    && DEFAULTS_GROUP_OPTION="" \
+    || DEFAULTS_GROUP_OPTION="--defaults-group-suffix='$WSREP_SST_OPT_CONF_SUFFIX'"
+
+    # Parent process already did the master key rotation
+    INVARIANT_OPTIONS="--binlog-rotate-encryption-master-key-at-startup=OFF"
+
+    wsrep_log_debug "-> CLONE_SOCK $CLONE_SOCK "
+    wsrep_log_debug "-> CLONE_LIBS $CLONE_LIBS "
+
+
+    DEFAULT_OPTIONS=" \
+     $DEFAULTS_FILE_OPTION \
+     $DEFAULTS_GROUP_OPTION \
+     $INVARIANT_OPTIONS \
+     --datadir='$WSREP_SST_OPT_DATA' \
+     --wsrep_on=0 \
+     --secure_file_priv='' \
+     --socket="$CLONE_SOCK" \
+     --log_error='$CLONE_ERR' \
+     --pid_file='$CLONE_PID_FILE' \
+     --plugin-dir='$CLONE_LIBS' \
+     --port='$JOINER_CLONE_PORT' \
+    "
+    SKIP_NETWORKING_OPTIONS=" \
+     --skip-networking \
+    "
+
+    # Define USER and PW
+    CLONE_PSWD=`wsrep_gen_secret`"$PSWD_POLICY"
+    CLONE_USER="clone_sst"
```

Comment:
If the SST fails, then the CLONE_PSWD is exposed in the error log as 

```
2025-02-12T13:47:00.653175Z 0 [Note] [MY-000000] [Galera] async IST sender starting to serve tcp://127.0.0.1:4004 sending 21-21, preload starts from 21
2025-02-12T13:47:00.653193Z 1 [ERROR] [MY-000000] [WSREP] Invalid sst_request: clone<nullptr>clone_sst:47aedb42ae36592bc14d33da451a8c0dAa@127.0.0.1:4002<nullptr>
2025-02-12T13:47:00.653201Z 1 [Warning] [MY-000000] [Galera] SST preparation failed
```

Reply:
fixed